src/test_driver/tizen/README.md

```diff
@@ -0,0 +1,61 @@
+# CHIP Tests on QEMU
+
+Tizen runs mostly on ARM architecture. In order to run tests on Tizen, we need
+to use QEMU. This document describes how to build and run CHIP tests on QEMU.
+
+## Obtaining Tizen QEMU Docker Image
+
+All tools and dependencies required to build and run tests on Tizen on QEMU are
+included in the `chip-build-tizen-qemu` docker image. One can pull the docker
+image from hub.docker.com or build it locally using the provided Dockerfile in
+`integrations/docker/images/chip-build-tizen-qemu` directory.
+
+```sh
+# Pull the image from hub.docker.com
+docker pull connectedhomeip/chip-build-tizen-qemu:latest
+```
+
+## Building and Running Tests on QEMU
+
+All steps described below should be done inside the docker container.
+
+```sh
+docker run -it --rm --name chip-tizen-qemu \
+    connectedhomeip/chip-build-tizen-qemu:latest /bin/bash
+```
+
+### Clone the connectedhomeip repository
+
+```sh
+git clone https://github.com/project-chip/connectedhomeip.git
+```
+
+### Activate the environment
+
+```sh
+cd connectedhomeip
+source scripts/activate.sh
+```
+
+### Build application variants that you would like to test
+
+```sh
+# Build chip-tool without BLE support (BLE is not supported on QEMU)
+./scripts/build/build_examples.py --target tizen-arm-chip-tool-no-ble build
+# Build lighting-app without BLE and WiFi support
+./scripts/build/build_examples.py --target tizen-arm-light-no-ble-no-wifi \
+    --enable-flashbundle build
+```
+
+### Generate and run test target
+
+```sh
+# Generate test target
+gn gen --check --fail-on-unused-args -root="$PWD/src/test_driver/tizen" --args="
+    target_os=\"tizen\" target_cpu=\"arm\"
+    tizen_chip_tool=\"$PWD/out/tizen-arm-chip-tool-no-ble/chip-tool\"
+    tizen_lighting_app_tpk=\"$PWD/out/tizen-arm-light-no-ble-no-wifi/package/out/org.tizen.matter.example.lighting-1.0.0.tpk\"
+" out/tizen-check
+# Run Time QEMU-based tests
+ninja -C out/tizen-check check
```

Comment:
This seems to be quite complicated given that it requires pre-building things.

Could we have the test target build what is required? Use expectation would be running `foo-bar-tests-qemu` would automatically run what is needed.

Reply:
Here I was trying to achieve some flexibility for running tests with a different app variants. The test driver code is the same, but one can feed different apps into the same test setup - exactly like with the `./scripts/tests/run_test_suite.py` where one has to pre-build all binaries used for testing. But maybe my reasoning is indeed wrong...

Initially we were trying to do exactly like you said - add example app targets as dependencies for the test target. Unfortunately, with such setup we've been facing some issues, like for example, duplicated target names (this can be resolved), or problem with setting up builds with different features (e.g. chip-tool with BLE support, but example app without it - I thought that maybe it would be nice to test interactions of apps built with different features enabled).

But maybe you are right. Maybe flexibility is not a good idea for automated testing :D I'll try to refactor it.

Reply:
I am thinking maybe the runner does not need to be encapsulated into `ninja check`.

That way one just builds things, and then has commands on how to run. We use that approach for integration tests: you build the apps, but have a test runner to make usage of the apps.

Reply:
> I am thinking maybe the runner does not need to be encapsulated into ninja check.

Yes, it is some solution for that, but also, like you've said few moths ago, it would be nice to have a standard way of running tests (without the requirement of prior knowledge about particular platform). Such solution is widely used by other build systems like autotools or cmake, where test target (e.g. `make check`) is generated by the build system itself. However, tests generated by build system are not flexible (and maybe should not be), so everyone who runs `make check` will get the same test result (providing that the test itself is not stochastic in nature).

In this PR the test target is generated by the `src/test_driver/tizen/BUILD.gn` but I was thinking about adding it to the main `/BUILD.gn`, so one could generate `--args='target_os="all"'` with optionally enabled some extra platforms (e.g. `enable_tizen_builds=true`), then runing tests would be as simple as typing `ninja check`.